src/index.js

```diff
@@ -56,48 +60,40 @@ function _renderToString(vnode, context, opts, inner, isSvgMode, selectValue) {
 		return '';
 	}
 
+	// #text nodes
+	if (typeof vnode !== 'object') {
+		return encodeEntities(vnode);
+	}
+
+	let pretty = opts.pretty,
+		indentChar = pretty && typeof pretty === 'string' ? pretty : '\t';
+
 	// wrap array nodes in Fragment
 	if (Array.isArray(vnode)) {
-		vnode = createElement(Fragment, null, vnode);
+		let rendered = '';
+		for (let i = 0; i < vnode.length; i++) {
+			if (pretty && i > 0) rendered += '\n';
+			rendered += _renderToString(
+				vnode[i],
+				context,
+				opts,
+				inner,
+				isSvgMode,
+				selectValue
+			);
+		}
+		return rendered;
 	}
 
 	let nodeName = vnode.type,
 		props = vnode.props,
 		isComponent = false;
-	context = context || {};
-	opts = opts || {};
-
-	let pretty = opts.pretty,
-		indentChar = pretty && typeof pretty === 'string' ? pretty : '\t';
-
-	// #text nodes
-	if (typeof vnode !== 'object' && !nodeName) {
-		return encodeEntities(vnode);
-	}
 
 	// components
 	if (typeof nodeName === 'function') {
 		isComponent = true;
 		if (opts.shallow && (inner || opts.renderRootComponent === false)) {
 			nodeName = getComponentName(nodeName);
-		} else if (nodeName === Fragment) {
-			let rendered = '';
-			let children = [];
-			getChildren(children, vnode.props.children);
-
-			for (let i = 0; i < children.length; i++) {
-				rendered +=
-					(i > 0 && pretty ? '\n' : '') +
-					_renderToString(
-						children[i],
-						context,
-						opts,
-						opts.shallowHighOrder !== false,
-						isSvgMode,
-						selectValue
-					);
-			}
-			return rendered;
```

Comment:
I think we should keep this branch here as a fast path for Fragments. It allows us to avoid allocating a component instance that we don't need since Fragments can't be stateful. This branch is hit for cases like this one:

```jsx
<div>
  <Fragment>
    <span />
  <Fragment />
</div>
```

Reply:
good point - added!